### PR TITLE
stm32: window watchdog: Fix interrupt priority for CM0 series and prescaler setup for new series.

### LINKED ISSUE #33000: stm32: window watchdog (wwdg): invalid interrupts priority for CM0 Series Socs
**Describe the bug**
All stm32 series define default wwdg interrupt priority of 7, but Cortex-M0 only have 2 bits for interrupt priority.

**To Reproduce**
1. enable wwdg.
2. execute watchdog sample or wdt_basic_api test.
3. Assertion fails when the priority is set during driver initialization(z_arm_irq_priority_set()).

**Impact**
Interrupt priority must be overridden locally, otherwise the application crashes.

### LINKED ISSUE #33001: stm32: window watchdog (wwdg): setup of prescaler not valid for newer series
**Describe the bug**
New Series (STM32G0, STM32G4, STM32H7, STM32L5, STM32WB, STM32WL) use a different window watchdog ip.
The programmable clock prescaler can be chosen in the range of 1 to 128(instead of 1 to 8).
Additionally, the used bit mask is not valid for these series.
This parameters are hardcoded, therefore the wwdg driver works only when the timeout is set to a value resulting in a prescaler value of 1.

**To Reproduce**
Execute the watchdog sample or wdt_basic_api test with the wwdg enabled and a timeout resulting in a prescaler value different from 1.

**Expected behavior**
The driver automatically calculates the appropriate prescaler value during setup.

**Impact**
WWDG timeout range is significantly reduced for the affected series.

### PR DESCRIPTION
**Pull request by str4t0m**
- Corrects invalid interrupt priority for F0, L0, and G0.
- Fixes prescaler bit mask for STM32G0, STM32G4, STM32H7, STM32L5, STM32WB, and STM32WL
- removes hardcoded prescaler range of 1 to 8.
- enable wwdg tests for nucleo_g431rb and nucleo_f091rc.

Fixes: #33000 
Fixes: #33001

### CODE REVIEW

**Review by FRASTM — APPROVED** [review]

File: drivers/watchdog/wdt_wwdg_stm32.c [hunk 1]
```diff
@@ -181,21 +201,22 @@ static int wwdg_stm32_install_timeout(const struct device *dev,
...
+	calculated_timeout = wwdg_stm32_get_timeout(dev, prescaler_exp, counter);
 
+	LOG_DBG("prescaler: %d", (1 << prescaler_exp));
 	LOG_DBG("Desired WDT: %d us", timeout);
 	LOG_DBG("Set WDT:     %d us", calculated_timeout);
 
-	if (!(IS_WWDG_PRESCALER(prescaler) && IS_WWDG_COUNTER(counter) &&
-	      IS_WWDG_TIMEOUT(timeout, calculated_timeout))) {
+	if (!(WWDG_COUNTER_IS_VALID(counter) &&
+	      WWDG_TIMEOUT_IS_VALID(timeout, calculated_timeout))) {
```
Comment (erwango) [inline drivers/watchdog/wdt_wwdg_stm32.c:219]: I would prefer we keep existing HAL marco.
I don't see really good reason to remove them, specially if this is to provide alternative implementation using a different name.

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for his contribution.

Looks fine aside 2 comments.

**Review by erwango — APPROVED** [review]
Nit: Commit message for "drivers/watchdog: wwdg stm32: fix prescaler setup for newer series." could deserve a slight rework.

File: samples/drivers/watchdog/boards/nucleo_g431rb.overlay [hunk 2]
```diff
@@ -0,0 +1,7 @@
+&wwdg {
```
[the remark this answers is not in the document]
Comment (str4t0m) [reply samples/drivers/watchdog/boards/nucleo_g431rb.overlay:7]: Per default the board uses a APB1 prescaler of 1 resulting in a ABP1 frequency of 170MHz.
WWDG has the fixed prescaler of 4096 and a variable prescaler of 1 to 128.
Therefore the valid range w/o a modified prescaler is:
t_WWDG_min = 1 / (170MHz / (4096 * 1)) = 24.1us.
t_WWDG_max = (0x3F + 1) / (170MHz / (4096 * 128)) = 197.4ms.

As the sample uses a max window of 100ms, this should be fine? What do I miss?
I don't have a nucleo_g431rb but I have tested it on a nucleo_g474re.

**Comment by str4t0m** [comment]
I have reworked the commit message regarding the "IS_WWDG_PRESCALER".

**Comment by str4t0m** [comment]
The complience check seems unrelated. I have rebased on the latest master, but that didn*t resolve the issue:

Found references to undefined Kconfig symbols.
...
CONFIG_DMA_0_NAME tests/drivers/dma/chan_link_transfer/src/test_dma.c:24

File: samples/drivers/watchdog/boards/nucleo_g431rb.overlay (hunk 2 above)
[the remark this answers is not in the document]
Comment (erwango) [reply samples/drivers/watchdog/boards/nucleo_g431rb.overlay:7]: That's fine indeed, txs.

**Pull request merged**